nixos/modules/services/databases/mysql.nix

```diff
@@ -476,6 +473,35 @@ in
               # ensureDatbases & ensureUsers depends on this script being run as root
               # when the user has secured their mysql install
               "+${setupScript}";
+          # User and group
+          User = cfg.user;
+          Group = "mysql";
+          # Runtime directory and mode
+          RuntimeDirectory = "mysqld";
+          RuntimeDirectoryMode = "0755";
+          # Access write directories
+          ReadWritePaths = [ cfg.dataDir ];
+          # Capabilities
+          CapabilityBoundingSet = "";
+          # Security
+          NoNewPrivileges = true;
+          # Sandboxing
+          ProtectSystem = "strict";
+          ProtectHome = mkDefault true;
```

Comment:
It might be best to take out `mkDefault` here and have the user explicitly force with `lib.mkForce`.

Reply:
Service nginx also uses this variant:
```
...
      } // optionalAttrs cfg.enableSandbox {
        # Sandboxing
        ProtectSystem = "strict";
        ProtectHome = mkDefault true;
...
```

Reply:
I don't see advantages of mkForce here. We only want to provide good defaults here but don't make it harder for users that have other use cases in mind.

Reply:
Changed to `ProtectHome = true;`

Reply:
My thought process is that whenever the user makes their system less secure we should do what we can to ensure they are aware of it. But yeah one way or the other, take your pick.